pyvrp/cpp/Measure.h

```diff
@@ -0,0 +1,118 @@
+#ifndef MEASURE_H
+#define MEASURE_H
+
+#include <compare>
+#include <type_traits>
+
+using value_type = int;
+
+enum class MeasureType
+{
+    DISTANCE,
+    DURATION,
+    COST,
+    CAPACITY,
+};
+
+template <MeasureType _> class Measure
+{
+    friend class Measure<MeasureType::DISTANCE>;
+    friend class Measure<MeasureType::DURATION>;
+    friend class Measure<MeasureType::COST>;
+    friend class Measure<MeasureType::CAPACITY>;
+
+    value_type value;
+
+public:
+    // Default construction initialises to 0.
+    Measure() = default;
+
+    // This constructor takes any arithmetic type (generally useful) and casts
+    // its value to value_type.
+    template <typename T,
+              std::enable_if_t<std::is_arithmetic_v<T>, bool> = true>
+    Measure(T const value) : value(static_cast<value_type>(value))
```

Comment:
Normally yes. But not for `std::max` and `std::min`, because those are templated with just one parameter, and thus implicitly require an explicit promotion [that was weird to write]. There's another way to do this now that I think about it more. I'll try that.